values/src/main/java/org/spine3/net/Urls.java

```diff
@@ -41,19 +53,43 @@ public static Url of(Url rawUrl) {
         return url;
     }
 
+    /**
+     * Converts String URL representation into {@link Url} instance.
+     *
+     * @param rawUrlString raw URL String
+     * @return {@link Url} with {@link org.spine3.net.Url.Record} instance
+     */
     public static Url of(String rawUrlString) {
         final Url.Builder builder = Url.newBuilder();
         builder.setRaw(rawUrlString);
         final Url rawUrl = of(builder.build());
         return rawUrl;
     }
 
+    /**
+     * Performs String conversion for given {@link Url}.
+     *
+     * @param url valid {@link Url} instance
+     * @return String representation of the given URL
+     * @throws IllegalArgumentException if the argument is invalid
+     */
     public static String toString(Url url) {
         validate(url);
         final String stringUrl = UrlPrinter.printToString(url);
         return stringUrl;
     }
 
+    /**
+     * Validates {@link Url} instance.
+     * <p>
+     * <p>{@link Url} with raw String is always valid.
+     * <br>{@link Url} with not set value is always invalid.
+     * {@link Url} can not have empty host. {@link org.spine3.net.Url.Record.Authorization} can't have
```

Comment:
Consider having rules enumerated in as unordered list. You may want to refer to RFC standard for URLs too.